### DISCUSSION TITLE
Add `sortObject` function

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#233] [Ideas]
Create a new object whose keys are sorted according to a given callback that compares object entries.

```ts
const newObject = sortObject(
  input,
  ([key1, value1], [key2, value2]) => {
    return key1 > key2 ? -1 : 1
  }
)
```

**Comment by Minhir** [comment]
What do you think about `sortObjectKeys` or `sortObjectEntries`? Sounds clearer to me.

**Reply by aleclarson** [reply]
How about `sortObjectByEntries`?

We could also have `sortObjectByKeys` which only passes in the key to the comparator function. The comparator could even be optional, which would use the same default behavior as Array `sort`.

**Comment by Minhir** [comment]
Oh, I think it's not possible to do it according to [spec](https://262.ecma-international.org/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys)

**Reply by aleclarson** [reply]
Sorry, what's not possible? You mean making it type-safe?

**Reply by Minhir** [reply]
> For each own property key P of O that is an integer index, in ascending numeric index order

It will override order of the one provided by comparator

**Reply by aleclarson** [reply]
Very interesting! Only for numeric keys though, so it's an edge case we'll mention in documentation.

**Reply by Minhir** [reply]
Sounds very fragile. I suggest to return `Map` instead of object.
